src/itemTypes/classifier.ts

```diff
@@ -113,28 +126,12 @@ export function initItemTemplateFromId (
             itemTemplate.data = dataResponse;
             itemTemplate.resources = resourceResponse && resourceResponse.total > 0 ? resourceResponse.resources : null;
 
-            // Complete item
-            const completionPromise = itemTemplate.fcns.completeItemTemplate(itemTemplate, requestOptions);
-
-            // Request item dependencies
-            const dependenciesPromise = itemTemplate.fcns.getDependencies(itemTemplate, requestOptions);
-
-            Promise.all([
-              completionPromise,
-              dependenciesPromise
-            ])
+            // Create the item's template
+            itemTemplate.fcns.convertItemToTemplate(itemTemplate, requestOptions)
             .then(
-              responses2 => {
-                const [completionResponse, dependenciesResponse] = responses2;
-                itemTemplate = completionResponse;
-
-                itemTemplate.dependencies = removeDuplicates(
-                  mCommon.deTemplatize(
-                    dependenciesResponse
-                    .reduce((acc, val) => acc.concat(val), [])  // some dependencies come out as nested, so flatten
-                  ) as string[]
-                );
-
+              template => {
+                itemTemplate.dependencies = // some dependencies come out as nested, so flatten
+                  removeDuplicates(flatten(template.dependencies));
                 resolve(itemTemplate);
               },
               () => reject({ success: false })
```

Comment:
Any issues if we allow potential system error to pass through these?

Something like:

(e) => reject({ success: false, error: e })

Reply:
This seems much more useful. Let's switch to that model for the future and I can retroactively update the older code.

Reply:
I tried this out in a few places to get more info on an error I am getting with the view and realized that this can lead to some nasty nesting of the actual message we are after.

![image](https://user-images.githubusercontent.com/2170893/53257570-bb28aa00-3687-11e9-83d6-45a0e2b811a3.png)

This seems to be a little better:

`(e) => reject({ success: false, error: e.error ? e.error : e })`

![image](https://user-images.githubusercontent.com/2170893/53257627-e3b0a400-3687-11e9-93b0-494c8acf6f98.png)

Reply:
Much better!